### ISSUE TITLE
API document for `x86_64-apple-darwin` is not shown on docs.rs

### ISSUE THREAD

**Issue by rhysd** [tauri-apps/wry#1032]
**Describe the bug**

https://docs.rs/wry/latest/wry/ doesn't show API documentation for `x86_64-apple-darwin` target. Users cannot know `*ExtMacOS` traits such as [`WebviewExtMacOS`](https://github.com/tauri-apps/wry/blob/cf34c2bde89e1bf1415902b4975ac71870af04c7/src/webview/mod.rs#L1174) by reading the documentation page.

**Steps To Reproduce**

1. Visit https://docs.rs/wry/latest/wry/
2. Find the 'Platform' pull down menu in menu bar on top of the page
3. Click the menu to open the list of supported targets

**Expected behavior**

`x86_64-apple-darwin` is listed in the pull down menu list.

**Screenshots**

<img width="468" alt="image" src="https://github.com/tauri-apps/wry/assets/823277/2d320daf-35b4-47d6-a449-6436bcfcf99a">

**Platform and Versions (please complete the following information):**
OS: N/A
Rustc: N/A

**Would you want to assign yourself to resolve this bug?**
- [ ] Yes
- [x] No

**Additional context**

`x86_64-apple-darwin` is listed in targets to build by docs.rs here.

https://github.com/tauri-apps/wry/blob/cf34c2bde89e1bf1415902b4975ac71870af04c7/Cargo.toml#L18-L22

The latest version was [built successfully on docs.rs](https://docs.rs/crate/wry/0.33.0/builds/909189). I could not understand why the target was not built.

I built the document locally on Ubuntu 22.04 (since API document for the platform is cross-compiled on Linux [as the document says](https://docs.rs/about/builds#cross-compiling)) and it was successful.

```sh
rustup target add x86_64-apple-darwin
cargo doc --target x86_64-apple-darwnin --features=dox,file-drop,protocol --no-default-features --open
```

**Comment by amrbashir** [comment]
@pewsheen, could you test the docs generation on your macOS and see why it fails?

**Comment by rhysd** [comment]
I tried to build the API doc on my macOS 13.6 as well. But the following also succeeded with exit status 0.

```sh
cargo doc --features=dox,file-drop,protocol --no-default-features
```

docs.rs cross-compiles the API document on a Linux container. It may cause some issue though the build log didn't report any error.

**Comment by rhysd** [comment]
I noticed [tauri docs](https://docs.rs/tauri/latest/tauri/) has this issue as well. I guess the cause is the same.

<img width="473" alt="image" src="https://github.com/tauri-apps/wry/assets/823277/1efabcb2-e571-4729-8165-8846dd7798ea">

**Comment by wusyong** [comment]
One possible reason might because we enabled [objc-exception](https://docs.rs/crate/objc_exception/latest).
IIRC, if dependencies couldn't be built in doc.rs, so is your crate.
Maybe we can try to remove it(or disable it by default). I find it didn't really help with debugging anyway.

**Comment by FabianLars** [comment]
seems like you're right, even though that feature is supposed to be disabled by default for docs.rs (via `disabled-features = false`) it seems to be able to cross compile after removing all mentions of that feature/dependency from cargo.toml

Edit: To be clear, i'm using the docs.rs build pipeline locally to test this as described in their readme https://github.com/rust-lang/docs.rs/blob/master/README.md#getting-started / https://github.com/rust-lang/docs.rs/blob/master/README.md#build-subcommand

![grafik](https://github.com/tauri-apps/wry/assets/30730186/cff026ad-7834-4663-9355-eb3e344727fe)

**Comment by FabianLars** [comment]
This feature would have been nice 😅 https://github.com/rust-lang/docs.rs/issues/787

**Comment by wusyong** [comment]
Closed by #1033

**Comment by rhysd** [comment]
Thank you for fixing this. I confirmed the API documents for `x86_64-apple-darwin` on docs.rs.
